partials/deprecated.html

```diff
@@ -0,0 +1,23 @@
+{% macro deprecated(channel, channelName) %}
+
+<a class="js-menu-item flex break-words no-underline text-grey-darkest mt-8 sm:mt-8 md:mt-3"
+    href="#operation-subscribe-{{ channelName }}">
+    {% if channel.deprecated %}
+    <span title="Deprecated"></span>
+    {% endif %}
```

Comment:
sorry but I just noticed this is some old stuff. AsyncAPI doesn't support `deprecated` property on a channel level and there is no way `channel.deprecated` returns something else than `undefined`.

please remove this part.

next steps would be to
1. rename this file to something like `tag_nav_item.html` and put it in the same dir as files like `by_tags.html `
2. your macro is created perfectly, without issues. Now just extend it a bit. Look, now, when you remove this `deprecated` part [this](https://github.com/asyncapi/html-template/blob/master/partials/operations/by_tags.html#L14-L53) seems to be duplicated twice in both files that you already modified. I think this `children` div  could entirely be moved into this macro. Correct me if I'm wrong

Reply:
Yeah you're right. Will add the entire div in the macro.

Reply:
There's a slight difference though in both the divs, [here](https://github.com/asyncapi/html-template/blob/11eb2b093275ac660dadec09ae9c519ae6d62b79/partials/operations/by_tags.html#L1-L2) and [here](https://github.com/asyncapi/html-template/blob/11eb2b093275ac660dadec09ae9c519ae6d62b79/partials/operations/by_tags_no_root.html#L1). The real difference in the macro would just be [here](https://github.com/asyncapi/html-template/blob/11eb2b093275ac660dadec09ae9c519ae6d62b79/partials/operations/by_tags_no_root.html#L16).
I think the only difference would be to give `categorytags` as an argument to the macro for `by_tags.html` and `tag` for `by_tags_no_root.html` Is that correct?

Reply:
I think we probably talk about different things here. I mean that this

```
<div class="children">
        {% for channelName, channel in asyncapi.channels() %}
           {% if channel.publish() and channel.publish().hasTags() and channel.publish() | containTags(categorytags) %}
            <a class="js-menu-item flex break-words no-underline text-grey-darkest mt-8 sm:mt-8 md:mt-3" href="#operation-publish-{{ channelName }}">
              <span class="bg-blue-dark font-bold h-6 no-underline text-white uppercase p-1 mr-2 rounded" style="height: 21px;font-size: 11px;" title="Publish">
                Pub
              </span>
              {% if channel.publish().summary() %}
                <span style="padding-top: 2px;">
                  {{ channel.publish().summary() }}
                </span>
              {% else %}
                <div style="display:inline-block;">
                  {{ slicedString(channelName, 'padding-top: 2px;') }}
                </div>
              {% endif %}
            </a>
          {% endif %}

          {% if channel.subscribe() and channel.subscribe().hasTags() and channel.subscribe() | containTags(categorytags) %}
            <a class="js-menu-item flex break-words no-underline text-grey-darkest mt-8 sm:mt-8 md:mt-3" href="#operation-subscribe-{{ channelName }}">
              {% if channel.deprecated %}
                <span title="Deprecated"></span>
              {% endif %}
              <span class="bg-green-dark font-bold no-underline text-white uppercase p-1 mr-2 rounded" style="height: 21px;font-size: 11px;" title="Subscribe">
                Sub
              </span>
              {% if channel.subscribe().summary() %}
                <span style="padding-top: 2px;">
                  {{ channel.subscribe().summary() }}
                </span>
              {% else %}
                <div style="display:inline-block;">
                  {{ slicedString(channelName, 'padding-top: 2px;') }}
                </div>
              {% endif %}
            </a>
          {% endif %}
        {% endfor %}
      </div>
```

looks like it is used in `by_tags.html` 2 times, and also 2 times in `by_tags_no_root.html`. And it could be moved to a single macro